FileFeedStoragePreFeedOptionsTest fails if /tmp/foobar already exists with different ownership

Quick one - if, for instance, you're running the the scrapy test suite in some sort of CI setup where builds get performed by random different dedicated "build users", you'll end up in a situation where `/tmp/foobar` already exists, owned by a different user. At which point `FileFeedStoragePreFeedOptionsTest` will fail:

```
E       PermissionError: [Errno 13] Permission denied: '/tmp/foobar'
```

Fairly trivial to fix - you could put a random suffix on the filename.. better, generate a directory to contain it using `tempfile.mkdtemp`.. many options.

Full build failure log: https://nix-cache.s3.amazonaws.com/log/5fdywg4w7s6cyf224j42yn9619kw5gw1-python3.8-Scrapy-2.4.1.drv

### Versions

Scrapy 2.4.1, python 3.8, macos 10.15

--- a/tests/test_feedexport.py
+++ b/tests/test_feedexport.py
@@ -1780,8 +1780,9 @@ class FileFeedStoragePreFeedOptionsTest(unittest.TestCase):
     maxDiff = None
 
     def test_init(self):
+        temp = tempfile.NamedTemporaryFile().name
         settings_dict = {
-            'FEED_URI': 'file:///tmp/foobar',
+            'FEED_URI': f'file:///{temp}',
             'FEED_STORAGES': {
                 'file': FileFeedStorageWithoutFeedOptions
             },

--- a/tests/test_feedexport.py
+++ b/tests/test_feedexport.py
@@ -1782,7 +1782,7 @@ class FileFeedStoragePreFeedOptionsTest(unittest.TestCase):
     def test_init(self):
         with tempfile.NamedTemporaryFile() as temp:
             settings_dict = {
-                'FEED_URI': f'file:///{temp.name}',
+                'FEED_URI': f'file://{temp.name}',
                 'FEED_STORAGES': {
                     'file': FileFeedStorageWithoutFeedOptions
                 },

--- a/tests/test_feedexport.py
+++ b/tests/test_feedexport.py
@@ -1780,9 +1780,9 @@ class FileFeedStoragePreFeedOptionsTest(unittest.TestCase):
     maxDiff = None
 
     def test_init(self):
-        with tempfile.NamedTemporaryFile().name as temp:
+        with tempfile.NamedTemporaryFile() as temp:
             settings_dict = {
-                'FEED_URI': f'file:///{temp}',
+                'FEED_URI': f'file:///{temp.name}',
                 'FEED_STORAGES': {
                     'file': FileFeedStorageWithoutFeedOptions
                 },

--- a/tests/test_feedexport.py
+++ b/tests/test_feedexport.py
@@ -1780,15 +1780,15 @@ class FileFeedStoragePreFeedOptionsTest(unittest.TestCase):
     maxDiff = None
 
     def test_init(self):
-        temp = tempfile.NamedTemporaryFile().name
-        settings_dict = {
-            'FEED_URI': f'file:///{temp}',
-            'FEED_STORAGES': {
-                'file': FileFeedStorageWithoutFeedOptions
-            },
-        }
-        crawler = get_crawler(settings_dict=settings_dict)
-        feed_exporter = FeedExporter.from_crawler(crawler)
+        with tempfile.NamedTemporaryFile().name as temp:
+            settings_dict = {
+                'FEED_URI': f'file:///{temp}',
+                'FEED_STORAGES': {
+                    'file': FileFeedStorageWithoutFeedOptions
+                },
+            }
+            crawler = get_crawler(settings_dict=settings_dict)
+            feed_exporter = FeedExporter.from_crawler(crawler)
         spider = scrapy.Spider("default")
         with warnings.catch_warnings(record=True) as w:
             feed_exporter.open_spider(spider)

--- a/tests/test_feedexport.py
+++ b/tests/test_feedexport.py
@@ -1782,7 +1782,7 @@ class FileFeedStoragePreFeedOptionsTest(unittest.TestCase):
     def test_init(self):
         with tempfile.NamedTemporaryFile() as temp:
             settings_dict = {
-                'FEED_URI': f'file://{temp.name}',
+                'FEED_URI': f'file:///{temp.name}',
                 'FEED_STORAGES': {
                     'file': FileFeedStorageWithoutFeedOptions
                 },

